packages/algorand-js/src/internal/interpreter/interpreter.ts

```diff
@@ -0,0 +1,33 @@
+import type { execParams } from "algob/src/types";
+import { assert } from "chai";
+
+import { IStack, Stack } from "../../lib/stack";
+import type { AppArgs, Operator, StackElem } from "../types";
+
+export class Interpreter {
+  readonly _stack: IStack<StackElem>;
+
+  constructor () {
+    this._stack = new Stack<StackElem>();
+  }
+
+  /**
+ * Description: this function executes set of Operator[] passed after
+ * parsing teal code
+ * @param {execParams} txn : Transaction parameters for current txn
+ * @param {Logic[]} logic : set of Logic objects
+ * @param {AppArgs} args : argument array passed to the current transaction
+ * @returns {boolean} : transaction accepted/rejected based on asc logic
+ */
+  execute (txn: execParams, logic: Operator[], args: AppArgs): boolean {
+    assert(Array.isArray(args));
+    for (const l of logic) {
+      l.execute(this._stack); // execute each teal opcode
+    }
+    if (this._stack.length() > 0) {
+      const top = this._stack.pop() as BigInt;
+      if (typeof top === 'string' || top >= BigInt("1")) { return true; } // Logic accept
```

Comment:
1. we can remove if and do just a return: `typeof top === 'string' || top >= BigInt("1")`
2. Why do we return true if top is a string? What if on the top we have empty string (`""`) or `"0"`?

Reply:
Sorry, you are correct. Updating

Reply:
Let's add tests for that.

Reply:
We need to be 100% sure that all this things work correctly.

Reply:
:+1: